src/storage/storage_structure/overflow_file.cpp

```diff
@@ -199,8 +188,8 @@ void OverflowFile::readFromDisk(transaction::TransactionType trxType, common::pa
 
 void OverflowFile::writePageToDisk(common::page_idx_t pageIdx, uint8_t* data) const {
     if (pageIdx < numPagesOnDisk) {
-        // TODO: updatePage does an unnecessary read + copy. We just want to overwrite
-        DBFileUtils::updatePage(*getBMFileHandle(), dbFileID, pageIdx, false, *bufferManager, *wal,
+        DBFileUtils::updatePage(*getBMFileHandle(), dbFileID, pageIdx,
```

Comment:
I feel we should rename the param `bool isInsertingNewPage` inside `DBFileUtils::updatePage`. The param is not intuitive. `bool readPageBeforeUpdate` should be better?

Reply:
Maybe. I'm hesitant to flip the meaning of the boolean though, as that's something easy to overlook. Maybe we should instead use the `BufferManager::PageReadPolicy` to make it more obvious what the parameter means?